### PR TITLE
BROOKLYN-350 Improving the download of external packages

### PR DESCRIPTION
**Pull request by ygy**
- externalizing the CURL command
- adding default parameter so curl follows location redirects

### CODE REVIEW

File: software/database/src/main/java/org/apache/brooklyn/entity/database/postgresql/PostgreSqlNode.java [hunk 1]
```diff
@@ -52,13 +52,17 @@
...
+    ConfigKey<String> SUGGESTED_VERSION = ConfigKeys.newConfigKeyWithDefault(SoftwareProcess.SUGGESTED_VERSION, "9.3-3");//"9.1-4");
 
     @SetFromFlag("configFileUrl")
     ConfigKey<String> CONFIGURATION_FILE_URL = ConfigKeys.newStringConfigKey(
             "postgresql.config.file.url", "URL where PostgreSQL configuration file can be found; "
                 + "if not supplied the blueprint uses the default and customises it");
 
+    @SetFromFlag("curlCommand")
+    ConfigKey<String> CURL_COMMAND = ConfigKeys.newStringConfigKey(
+            "curl.command", "CURL comamnd with parameters to be used", "curl -L");
```
Comment (bostko) [inline software/database/src/main/java/org/apache/brooklyn/entity/database/postgresql/PostgreSqlNode.java]: Could you rename it to `cURL`.
Could you try adding more info on this parameter?

**Comment by bostko** [comment]
@ygy Not very intuitive approach with using a ConfigKey. Still could be sensible in some situations.

File: software/database/src/main/java/org/apache/brooklyn/entity/database/postgresql/PostgreSqlNode.java [hunk 2]
```diff
@@ -52,13 +52,17 @@
...
+    ConfigKey<String> SUGGESTED_VERSION = ConfigKeys.newConfigKeyWithDefault(SoftwareProcess.SUGGESTED_VERSION, "9.3-3");//"9.1-4");
 
     @SetFromFlag("configFileUrl")
     ConfigKey<String> CONFIGURATION_FILE_URL = ConfigKeys.newStringConfigKey(
             "postgresql.config.file.url", "URL where PostgreSQL configuration file can be found; "
                 + "if not supplied the blueprint uses the default and customises it");
 
+    @SetFromFlag("curlCommand")
+    ConfigKey<String> CURL_COMMAND = ConfigKeys.newStringConfigKey(
+            "curl.command", "cURL comamnd to be used. It follows location redirects by default (added -L parameter)", "curl -L");
```
Comment (bostko) [inline software/database/src/main/java/org/apache/brooklyn/entity/database/postgresql/PostgreSqlNode.java]: Strange config usage of а parameter.
Very important to add a sentence about checking the parameter usage in Java code.

I would rather declare a CURL_COMMAND_OPTIONS parameter and put the single `-L` as a default value.

Comment (neykov) [inline software/database/src/main/java/org/apache/brooklyn/entity/database/postgresql/PostgreSqlNode.java]: +1 could even do away with the config altogether - that's not something I've ever seen needing configuration. No strong opinion though.

**Comment by ygy** [comment]
@bostko @neykov Can you review the new version?

**Comment by bostko** [comment]
@ygy Looks good!

**Comment by neykov** [comment]
+1, merging.

**Pull request merged**